### PR TITLE
Node.js week3 homework - calculator

### PR DESCRIPTION
**Pull request by shpomp**
Warmup - calculator

### CODE REVIEW

File: nodejs/week3/warmup/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          params: extractQueryNumbers(object),
+          result:
+            Math.round(extractQueryNumbers(object).reduce(action) * 100) / 100,
+        };
+  console.log(response);
+  return response;
+};
+
+const add = (accumulator, currentValue) => accumulator + currentValue;
+const subtract = (accumulator, currentValue) => accumulator - currentValue;
```
Comment (cecastosic) [inline nodejs/week3/warmup/index.js:38]:
```suggestion
const substract = (accumulator, currentValue) => accumulator - currentValue;
```

File: nodejs/week3/warmup/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+const multiply = (accumulator, currentValue) => accumulator * currentValue;
+const divide = (accumulator, currentValue) => accumulator / currentValue;
+
+// -- ADD
+app.get("/add", (req, res) => {
+  console.log("add");
```
Comment (cecastosic) [inline nodejs/week3/warmup/index.js:44]: I would remove all console.log(...)

File: nodejs/week3/warmup/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+const app = express();
+
+app.get("/", (req, res) => res.send("nodejs week3 homework"));
+
+// www.javascripttutorial.net/object/convert-an-object-to-an-array-in-javascript/
+// https://stackoverflow.com/questions/32817027/check-if-an-array-contains-only-numeric-values
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce
+
+const extractQueryNumbers = (object) => {
```
Comment (cecastosic) [inline nodejs/week3/warmup/index.js:12]: Really cool 💪

File: nodejs/week3/warmup/index.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  const unnestQueryNumbers = queryNumbers.flat();
+  const numbers =
+    unnestQueryNumbers.some(isNaN) || unnestQueryNumbers.length < 2
+      ? []
+      : unnestQueryNumbers.map((i) => Number(i));
+  return numbers;
+};
+
+// repeating code of each path into a function:
+const processQuery = (action, object) => {
```
Comment (cecastosic) [inline nodejs/week3/warmup/index.js:23]: Also cool 💪

File: nodejs/week3/warmup/index.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+// www.javascripttutorial.net/object/convert-an-object-to-an-array-in-javascript/
+// https://stackoverflow.com/questions/32817027/check-if-an-array-contains-only-numeric-values
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce
+
+const extractQueryNumbers = (object) => {
+  const queryNumbers = Object.values(object);
+  const unnestQueryNumbers = queryNumbers.flat();
+  const numbers =
+    unnestQueryNumbers.some(isNaN) || unnestQueryNumbers.length < 2
```
Comment (cecastosic) [inline nodejs/week3/warmup/index.js:16]: Nice that you thought about what should happen if the query parameters are not numbers!

File: nodejs/week3/warmup/readme.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+Calculator
```
Comment (cecastosic) [inline nodejs/week3/warmup/readme.md:1]: Nice with readme. 👍

File: nodejs/week3/warmup/readme.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+Getting data through query parameters using GET
+Going to /calculator/multiply?firstParam=1&secondParam=2 should respond with 2.
+Going to /calculator/multiply?firstParam=1&secondParam=2&secondParam=4 should respond with 8.
+Going to /calculator/add?firstParam=1&secondParam=2&secondParam=4 should respond with 7.
+There can be an infinite number of query parameters!
+
+Getting data through the request body using POST
+
+Going to /calculator/multiply with the following key values: firstParam=1, secondParam=2. Should respond with 2.
+Going to /calculator/division with the following key values: firstParam=1, secondParam=2. Should respond with 0.5.
```
Comment (cecastosic) [inline nodejs/week3/warmup/readme.md:21]: This part is missing from the homework. Have you tried to make a request using Postman? https://github.com/HackYourFuture-CPH/node.js/blob/master/week3/homework-template/readme.md#getting-data-through-the-request-body-using-post

**Review by cecastosic — COMMENTED** [review]
Great job! 💪 
All routes should be with /calculator, for example /calculator/multiply?firstParam=1&secondParam=2, instead of /multiply?firstParam=1&secondParam=2

File: nodejs/week3/warmup/index.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+app.get("/", (req, res) => res.send("nodejs week3 homework"));
+
+// www.javascripttutorial.net/object/convert-an-object-to-an-array-in-javascript/
+// https://stackoverflow.com/questions/32817027/check-if-an-array-contains-only-numeric-values
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce
+
+const extractQueryNumbers = (object) => {
+  const queryNumbers = Object.values(object);
+  const unnestQueryNumbers = queryNumbers.flat();
```
Comment (cecastosic) [inline nodejs/week3/warmup/index.js:14]:
```suggestion
  const unnestQueryNumbers = Object.values(object).flat();
```
a little bit shorter

File: nodejs/week3/warmup/readme.md (hunk 7 above)
Comment (shpomp) [reply nodejs/week3/warmup/readme.md:21]: Missed that part somehow! 🙈

**Comment by shpomp** [comment]
Thank you very much @cecastosic  for your time and the feedback! ❤️  ⭐

**Pull request merged**